var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -108,6 +107,13 @@ class PyScipy(PythonPackage):
     depends_on("lapack")
     depends_on("blas")
 
+    # Historical dependencies
+    depends_on("meson", when="@1.9.0:1.9.1", type="build")
+    depends_on("py-setuptools", when="@:1.8", type="build")
+    depends_on("py-setuptools@:59", when="@1.8", type="build")
+    depends_on("py-setuptools@:57", when="@1.7", type="build")
+    depends_on("py-setuptools@:51.0.0", when="@1.6", type="build")
```

Comment:
Not sure which of these upper bounds are strict, but I do remember some of these being required.

Reply:
Yep, these I would not touch. `setuptools` 60 was definitely breaking everything, not sure about the exact upper bounds for the rest.